### DISCUSSION TITLE
Can't load js file in bundle

### DISCUSSION THREAD

**Discussion by CaptnChaos-dev** [pimcore/pimcore#17715] [Q&A / Support] [answered]
Loading failed for the <script> with the source ‘http://localhost/admin/bundles/Divae/ProductImporterBundle/public/js/pimcore/startup.js’.

The location of the bundle is  /bundles/Divae/ProductImporterBundle/ 

I have done everything as described in the documentation. I created the bundle with the bundle generator. 
Does anyone have an idea what the problem could be?

**Comment by jdreesen** [comment]
Please show the code where you register the file. The path is wrong.

**Comment by CaptnChaos-dev** [comment]
I have changed the path:

Loading failed for the <script> with source “http://localhost/bundles/Divae/src/ProductImporterBundle/Resources/public/js/startup.js”.

service.yaml
```
Divae\ProductImporterBundle\EventListener\AdminAssetsListener:
        tags:
            - { name: kernel.event_listener, event: pimcore.bundle_manager.paths.js, method: onJsPaths }

```
            
AdminAssetListener
```
public static function getSubscribedEvents(): array
    {
        return [
            BundleManagerEvents::JS_PATHS => 'onJsPaths'
        ];
    }

    public function onJsPaths(PathsEvent $event):  void
    {
        $event->addPaths([
           '/bundles/Divae/src/ProductImporterBundle/Resources/public/js/startup.js',
           '/bundles/Divae/src/ProductImporterBundle/Resources/public/js/admin.js',
        ]);
    }
          
```
            
ProductImporterBundel    
```
   /**
     * @return string[]
     */
    public function getJsPaths(): array
    {
        return [
            '/bundles/Divae/src/ProductImporterBundle/Resources/public/js/startup.js',
            '/bundles/Divae/src/ProductImporterBundle/Resources/public/js/admin.js',
        ];
    }
        
```
    
    I hope this will help.

**Reply by jdreesen** [reply, marked as answer]
The assets need to be in the `public` folder of your project (the “web root”) otherwise you can't easily access them via the browser.

This happens by running `console assets:install` (in dev mode, preferably also with the `--symlink --relative` flags, because otherwise you have to call it again after each change).

This copies (or symlinks) everything from `bundles/Divae/src/ProductImporterBundle/Resources/` (if this is where your bundle lives) to `public/bundles/divaeproductimporter/` (the last part depends on your actual bundle name, so I just made a guess — please take a look into your `public/bundles/` folder if it is right).

And this is the path you must also use in your `AdminAssetListener` or `ProductImporterBundle` class (and by the way: one of the methods is enough, otherwise you'll add them twice):
```
[
   '/bundles/divaeproductimporter/js/startup.js',
   '/bundles/divaeproductimporter/js/admin.js',
]
```

**Reply by CaptnChaos-dev** [reply]
Thanks, now it works. But the path is 
```
[
   '/bundles/productimporter/js/startup.js',
   '/bundles/productimporter/js/admin.js',
]
```
